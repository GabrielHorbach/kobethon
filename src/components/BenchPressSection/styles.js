import styled from 'styled-components';

export const Container = styled.section`
  background: #4E5B82;
  padding: 6rem 10rem;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  height: calc(100% - 7rem);

  @media (max-width: 1200px) {
    padding: 6rem 3rem;
  }

  @media (max-width: 650px) {
    height: calc(100vh - 7rem);
    padding: 0;
    flex-direction: column;
  }
`;

export const VideoWrapper = styled.div`
  width: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media (max-width: 650px) {
    margin: 0 auto;
  }

  video {
    width: 90%;
  }

  div {
    width: 82%;
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: space-between;

    @media (max-width: 950px) {
      flex-direction: column;
    }

    @media (max-width: 650px) {
      align-items: center;
    }

    p {
      color: #fff;
      display: flex;
      align-items: center;
      font-size: 1rem;

      @media (max-width: 1200px) {
        margin-top: 1.2rem;  
      }

      svg {
        margin-right: .5rem;
      }
    }

    button {
      width: auto;
      border: none;
      padding: 1rem 1.5rem;
      background:#75C46B;
      color: #fff;
      border-radius: 5px;
      font-size: 1rem;
      transition: all ease .2s;
      opacity: .5;

      @media (max-width: 900px) {
        padding: .8rem 1.3rem;
      }
 
      @media (max-width: 650px) {
        margin-top: 1rem;
      }

      &:hover {
        opacity: 1;
        cursor: pointer;
      }
    }
  }
`;

export const InfoWrapper = styled.article`
  width: 50%;
  height: auto;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin-top: 3rem;
  margin-left: 3rem;

  @media (max-width: 800px) {
    margin-top: 2rem;
  }

  @media (max-width: 650px) {
    width: 100%;
    flex-direction: column;
    margin: 3rem auto 0;
    padding: 1rem 3rem 0;
    align-items: flex-start;
  }

  p {
    color: #fff;
    font-size: 1.5rem;
    line-height: 2.5rem;

    @media (max-width: 1100px) {
      font-size: 1.3rem;
      line-height: 2.3rem;
    }

    @media (max-width: 800px) {
      font-size: 1.2rem;
      line-height: 2.2rem;
    }
  }
`;
