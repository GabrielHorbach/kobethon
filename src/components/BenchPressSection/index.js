/* eslint-disable jsx-a11y/media-has-caption */
import React, { useState, useRef } from 'react';

import { FaCheckCircle } from 'react-icons/fa';

import BenchPress from '../../assets/animations/bench_press.mp4';

import { Container, VideoWrapper, InfoWrapper } from './styles';

export default function BenchPressSection() {
  const [reps, setReps] = useState(7);
  const videoRef = useRef(null);

  const myInformations = [
    <p>
      Mas agora vamos ficar menos sérios! Role para baixo para continuar.
    </p>,
    <p>
      Hoje minha stack é composta toda por JavaScript, onde eu utilizo ReactJS no frontend,
      NodeJS no backend e estou iniciando no desenvolvimento mobile com React Native.
    </p>,
    <p>
      Nesse meio tempo comecei a me aventurar pelo JavaScript e foi aí que eu decidi: era
      com isso que eu queria trabalhar dali em diante.
      </p>,
    <p>
      Liderei o desenvolvimento de um deles, um ERP grande e complexo, além de ter auxiliado em um
      app para Frente de Caixa e outros projetos internos.
    </p>,
    <>
      <p>
        Falando sobre a carreira, iniciei no final de 2015 um estágio onde posteriormente fui
        efetivado como desenvolvedor PHP trabalhando em softwares voltados a gestão de
        supermercados.
      </p>
    </>,
    <>
      <p>
        Entrei em Engenharia da Computação para dois semestres depois ver que aquilo não tinha nada
        a ver com o que eu queria.
      </p><br />
      <p>
        Foi aí que mudei para Análise e Desenvolvimento de Sistemas, que estou cursando atualmente.
      </p>
    </>,
    <>
      <p>
        Comecei os estudos de programação em 2014 no Curso Técnico da QI. Lá eu percebi que
        realmente gostava da coisa e quis me aprofundar.
      </p><br />
      <p>
        E aparentemente eu fui pelo caminho errado.
      </p>
    </>,
    <>
      <p>
        Me chamo Gabriel, tenho 23 anos e sou desenvolvedor web, apaixonado por tecnologia,
        música e bodybuilding (:
      </p><br />
      <p>
        Complete a série de repetições clicando no botão verde.
      </p>
    </>,
  ];

  function doRepetition(event) {
    event.preventDefault();

    if (reps === 0) return;

    videoRef.current.playbackRate = 2;
    videoRef.current.play();
    setReps(reps - 1);
  }

  function repetitionCounter() {
    if (reps > 1) {
      return <p>Restam {reps} repetições!</p>;
    }

    if (reps > 0) {
      return <p>Resta {reps} repetição!</p>;
    }

    return <p><FaCheckCircle size="2em" />Muito bem!</p>;
  }

  return (
    <Container>
      <VideoWrapper>
        <video ref={videoRef}>
          <source src={BenchPress} type="video/mp4" />
        </video>

        <div>
          {reps > 0
            && (
              <button type="button" onClick={doRepetition}>
                Fazer repetição
              </button>
            )}
          {repetitionCounter()}
        </div>
      </VideoWrapper>
      <InfoWrapper>
        {myInformations[reps]}
      </InfoWrapper>
    </Container>
  );
}
