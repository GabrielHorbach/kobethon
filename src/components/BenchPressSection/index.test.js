import React, { useRef } from 'react';

import { shallow } from 'enzyme';

import BenchPressSection from './index';

jest.mock('react', () => {
  const originReact = jest.requireActual('react');
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef,
  };
});

describe('Bench Press Section', () => {
  const wrapper = shallow(<BenchPressSection />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should play video when user clicks on button', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((init) => [init, setState]);

    const mRef = { current: { play: jest.fn() } };
    useRef.mockReturnValueOnce(mRef);
    const wrapper = shallow(<BenchPressSection />);

    wrapper.find('button').simulate('click', {
      preventDefault: () => { }
    });

    expect(mRef.current.play).toHaveBeenCalled();
  });
});