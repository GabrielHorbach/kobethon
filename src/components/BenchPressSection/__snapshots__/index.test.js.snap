// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bench Press Section should render correctly 1`] = `ShallowWrapper {}`;
